server/src/sessionManager.ts

```diff
@@ -373,10 +437,12 @@ async function installPyright(requestedVersion: string | undefined): Promise<Ins
     }
 
     return new Promise<InstallPyrightInfo>((resolve, reject) => {
-        const dirName = `./pyright_local/${version}`;
+        const dirName = path.join(pyrightLocalDir, version);
 
         if (fs.existsSync(dirName)) {
             logger.info(`Pyright version ${version} already installed`);
+            recordPyrightVersionUse(version);
+            prunePyrightCache();
```

Comment:
I don't think we need to call prunePyrightCache in this case because no new version was downloaded.